<launch>
   <node name="joy_node" pkg="joy" type="joy_node" required="true">
     <param name="dev" value="/dev/input/js0"/>
  </node>
  <!--<group ns="mambo_wps">-->
  <rosparam command="load" file="$(find Mambo_ROS_Examples)/flights/test_time_rhinf.yaml"/>
  <arg name="bt_addr" default="e0:14:8D:47:3d:c7"/>
  <!--<<node pkg="ros_pyparrot" type="mambo.py" name="mambo" output="screen">
    <param name="bt" value="$(arg bt_addr)"/>
  </node>-->
  <node pkg="Mambo_ROS_Examples" type="mambo_waypoints_node" name="mambo_waypoints_1" output="screen" required="true"/>
  <!--<node pkg="Mambo_ROS_Examples" type="mambo_waypoints_time" name="mambo_waypoints_time_1" output="screen" required="true"/>-->
  <node pkg="pid" type="controller" name="position_X">
    <remap from="setpoint" to="setpoint_pos_X"/>
    <remap from="state" to="state_pos_X"/>
    <remap from="control_effort" to="control_effort_pos_X"/>
    <param name="Kp" value="0.85" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.6" />
    <param name="upper_limit" value="1.00" />
    <param name="lower_limit" value="-1.00" />
    <param name="windup_limit" value="1" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="position_Y">
    <remap from="setpoint" to="setpoint_pos_Y"/>
    <remap from="state" to="state_pos_Y"/>
    <remap from="control_effort" to="control_effort_pos_Y"/>
    <param name="Kp" value="0.85" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.6" />
    <param name="upper_limit" value="1.00" />
    <param name="lower_limit" value="-1.00" />
    <param name="windup_limit" value="1" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="altittude">
    <remap from="setpoint" to="setpoint_alt"/>
    <remap from="state" to="state_alt"/>
    <remap from="control_effort" to="control_effort_alt"/>
    <param name="Kp" value="3.5" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.0001" />
    <param name="upper_limit" value="1.00" />
    <param name="lower_limit" value="-1.00" />
    <param name="windup_limit" value="1" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="heading">
    <remap from="setpoint" to="setpoint_hdg"/>
    <remap from="state" to="state_hdg"/>
    <remap from="control_effort" to="control_effort_hdg"/>
    <param name="Kp" value="0.02" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.005" />
    <param name="upper_limit" value="1.00" />
    <param name="lower_limit" value="-1.00" />
    <param name="windup_limit" value="1" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <!--<node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />-->
  <node pkg="rosbag" type="record" name="rosbag_record_drone"
       args="record -o $(find Mambo_ROS_Examples)/Bags/test_PD /vicon/Mambo_5/Mambo_5 /control_effort_alt /control_effort_hdg /control_effort_pos_X /control_effort_pos_Y /setpoint_pos_X /setpoint_pos_Y /setpoint_alt /r_setpoint_hdg /ready /record /hdg /rhinf_ref_x /rhinf_st_x /rhinf_ref_y /rhinf_st_y" />
  <!--</group>-->
</launch>
